From ab37f2442c27907e0e21edae6d36c5dead08ab6d Mon Sep 17 00:00:00 2001
From: Josua Mayer <josua@solid-run.com>
Date: Mon, 8 Jul 2024 11:53:42 +0200
Subject: [PATCH 21/21] board: sr_rzg2l: unconditionally flush i2c bus before
 reading eeprom

Memory configuration is critical and should always succeed.
Flush i2c bus before accessing eeprom to ensure the bus is not stuck.

Signed-off-by: Josua Mayer <josua@solid-run.com>
---
 plat/renesas/rz/board/sr_rzg2l/ddr_params.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/plat/renesas/rz/board/sr_rzg2l/ddr_params.c b/plat/renesas/rz/board/sr_rzg2l/ddr_params.c
index a4a874d49..922e45645 100644
--- a/plat/renesas/rz/board/sr_rzg2l/ddr_params.c
+++ b/plat/renesas/rz/board/sr_rzg2l/ddr_params.c
@@ -103,6 +103,7 @@ void ddr_param_setup(void)
 	cpg_i2c_setup((1 << 1));
 	pfc_i2c01_setup(0, 1);
 	riic_setup(RZG2L_RIIC1_BASE);
+	riic_flush(RZG2L_RIIC1_BASE);
 	status = riic_tlv_read_eeprom(0x50, 0x00);
 	pfc_i2c01_setup(0, 0);
 	cpg_i2c_setup(0);
-- 
2.35.3

